File: Python/0001_two_sum.py
# 2020-08-04 00:01:32

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        _map = {}
        for index, num in enumerate(nums):
            diff = target - num
            if diff in _map:
                return [_map[diff], index]
            else:
                _map[num] = index
        return [-1, -1]

File: C++/0001_twosum.cc
// 2020-08-04 00:44:13

#include <vector>
#include <unordered_map>

using std::unordered_map;
using std::vector;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if (m.count(diff))
                return {m[diff], i};
            m[nums[i]] = i;
        }
        return {};
    }
};
